### PR TITLE
[ttnn] Apply Bessel's correction to batch_var in batch_norm running_statistics update (#55438)

### LINKED ISSUE #55438: [ttnn/ops]: ttnn.batch_norm updates running_var with biased variance instead of unbiased estimator (missing Bessel's correction)
### Component / Area

ops, ttnn normalization

### Issue Type (optional)

Bad Outputs

#### **Observed:**
In training mode (`training=True`), `ttnn.batch_norm` updates `running_var` using the biased sample variance ($\frac{1}{m} \sum_{i=1}^m (x_i - \mu)^2$) computed directly by `mean_NHW`, without applying Bessel's correction factor $\frac{m}{m - 1}$:

```cpp
// ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm.cpp:109-135
batch_mean = operations::normalization::mean_NHW(input, memory_config, compute_kernel_config);
auto centered = ttnn::subtract(input, batch_mean, std::nullopt, memory_config);
batch_var = operations::normalization::mean_NHW(
    ttnn::square(centered, memory_config), memory_config, compute_kernel_config);
...
if (training && (running_mean.has_value() || running_var.has_value())) {
    ttnn::prim::running_statistics(
        batch_mean, batch_var, momentum, running_mean, running_var, memory_config, compute_kernel_config);
}
```

Because Bessel's correction is omitted, `running_var` systematically under-estimates the population variance by a factor of $\frac{m - 1}{m}$ on every training iteration. For instance, with small spatial dimensions or small batch sizes (e.g., $m = N \times H \times W = 8$), `running_var` is updated with a value that is 5.0% to 14.3% lower than PyTorch on every step.

#### **Expected:**
According to the PyTorch reference specification (`torch.nn.BatchNorm2d` and `torch.nn.functional.batch_norm`):

> *"At train time in the forward pass, the standard-deviation is calculated via the biased estimator, equivalent to `torch.var(input, correction=0)`. However, the value stored in the moving average of the standard-deviation is calculated via the **unbiased estimator**, equivalent to `torch.var(input, correction=1)`."*

While the current batch normalization output must use the biased sample variance $\sigma_B^2$, the moving average update for `running_var` must be scaled by Bessel's correction factor:

$$\text{unbiased\_var} = \text{batch\_var} \times \frac{m}{m - 1}$$
$$\text{running\_var} \leftarrow (1 - \text{momentum}) \times \text{running\_var} + \text{momentum} \times \text{unbiased\_var}$$

where $m = N \times H \times W$.

---

#### **1. Steps (exact commands):**
Run the following standalone script demonstrating the numerical divergence between PyTorch's `running_var` update and TTNN's current formula:

```python
import torch
import torch.nn as nn

# Sample input: Batch=2, Channel=1, Height=2, Width=2 -> total m = 2*2*2 = 8 elements per channel
x = torch.tensor([[[[1.0, 2.0],
                    [3.0, 4.0]]],
                  [[[5.0, 6.0],
                    [7.0, 8.0]]]], dtype=torch.float32)

momentum = 0.1
initial_var = 1.0

# 1. PyTorch Reference Behavior (uses correction=1 for running_var update)
bn = nn.BatchNorm2d(num_features=1, momentum=momentum, affine=False)
bn.running_mean.zero_()
bn.running_var.fill_(initial_var)
_ = bn(x)
pytorch_updated_var = bn.running_var.item()

# 2. TTNN Current Implementation (passes biased batch_var with correction=0 directly)
m = x.shape[0] * x.shape[2] * x.shape[3]  # m = 8
biased_var = torch.var(x, correction=0).item()   # 5.2500
unbiased_var = torch.var(x, correction=1).item() # 6.0000 (5.25 * 8 / 7)

ttnn_current_var = (1 - momentum) * initial_var + momentum * biased_var

print(f"Number of elements per channel (m = N*H*W): {m}")
print(f"Biased sample variance (correction=0):      {biased_var:.4f}")
print(f"Unbiased sample variance (correction=1):    {unbiased_var:.4f}")
print(f"Bessel's correction factor (m / (m - 1)):   {m / (m - 1):.4f}")
print("-" * 55)
print(f"PyTorch updated running_var:                {pytorch_updated_var:.4f}")  # 1.5000
print(f"TTNN current updated running_var:           {ttnn_current_var:.4f}")     # 1.4250
print(f"Discrepancy:                                {abs(pytorch_updated_var - ttnn_current_var):.4f} (5.00% error)")
```

#### **2. Input data / link or description:**
```text
Synthetic 4D tensor of shape [2, 1, 2, 2] (or any 4D tensor with finite batch and spatial dimensions).
```

#### **3. Frequency:**
```text
Always (Deterministic mathematical divergence on every training step).
```

---

#### **1. Software Versions:**
```text
OS: Ubuntu 22.04 / Linux / Windows
tt-metal: main branch
PyTorch: 2.1+
```

#### **2. Hardware Details:**
```text
All Tenstorrent Architectures (Host composite preprocessing logic in batch_norm.cpp before device dispatch).
```

---

### Is this a regression?

Unknown

### Regression Details

_No response_

### Logs & Diagnostics

_No response_

### Priority

P2

### Proposed Fix (Reference Diff)

In `ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm.cpp`:

```diff
     // Skip when neither running stat is provided: the return value is discarded.
     if (training && (running_mean.has_value() || running_var.has_value())) {
-        ttnn::prim::running_statistics(
-            batch_mean, batch_var, momentum, running_mean, running_var, memory_config, compute_kernel_config);
+        Tensor var_for_running_stats = batch_var;
+        if (running_var.has_value()) {
+            const auto& shape = input.logical_shape();
+            const float m = static_cast<float>(shape[0] * shape[-2] * shape[-1]);
+            if (m > 1.0f) {
+                const float bessel_correction = m / (m - 1.0f);
+                var_for_running_stats = ttnn::multiply(batch_var, bessel_correction, std::nullopt, memory_config);
+            }
+        }
+        ttnn::prim::running_statistics(
+            batch_mean, var_for_running_stats, momentum, running_mean, running_var, memory_config, compute_kernel_config);
     }

     return output_tensor;
```

### PR DESCRIPTION
**Pull request by MusounoEnma**
### Ticket
Fixes #55438

### Problem description
In `ttnn.batch_norm` (`training=True`), `running_var` was updated using the biased sample variance (`batch_var` computed via `mean_NHW`) without scaling by Bessel's correction factor $\frac{m}{m - 1}$. 

According to the PyTorch reference specification (`torch.nn.BatchNorm2d` and `torch.nn.functional.batch_norm`), while the current batch normalization output uses biased variance (`correction=0`), the moving average update for `running_var` must use the unbiased estimator (`correction=1`). Without this factor, `running_var` systematically under-estimates the population variance by 5% to 14% on small batches/spatial shapes, degrading model evaluation accuracy during inference (`training=False`).

### What's changed
Scale `batch_var` by $\frac{m}{m - 1}$ before forwarding it into `ttnn::prim::running_statistics` when `running_var` is present and $m > 1$.

### Checklist
- [x] Verified numerical parity against PyTorch reference
- [x] Retains original biased variance for current batch normalization
- [x] Fixes #55438

### CODE REVIEW

**Review by bbradelTT — COMMENTED** [review]
Looks good.

Please add a ttnn unit test that fails without this change and passes with this change.

**Comment by MusounoEnma** [comment]
@bbradelTT Thanks for the review

I have added the requested unit test `test_batch_norm_running_var_bessel_correction` in `tests/ttnn/unit_tests/operations/fused/test_batch_norm.py`.

### Failure vs. Pass Behavior:
- **Without this change:** `ttnn.batch_norm` updates `running_var` using the biased sample variance (`1.425` vs. PyTorch `1.500`), producing a ~5.0% relative discrepancy that **fails** the `rtol=0.01` tolerance assertion.
- **With this change:** Bessel's correction factor $\frac{m}{m - 1} = \frac{8}{7} \approx 1.1429$ correctly scales the batch variance, restoring exact numerical parity with PyTorch (`1.500`), and the test cleanly **passes**.

**Comment by MusounoEnma** [comment]
Hi @bbradelTT, just checking in on this PR since it's now tagged `pr-priority:p1` / `pr-risk:high`. The requested unit test has been added and is passing. Let me know if there's anything else needed before this can be approved and merged. Thanks!

**Review by bbradelTT — APPROVED** [review]

**Comment by MusounoEnma** [comment]
Thank you @bbradelTT and @mateusznowakTT for the review and merge! 

Could you please let me know if this fix is eligible for a community bounty reward, or if there are upcoming bounty issues I could be assigned to work on next? 

Thanks again!

**Comment by mateusznowakTT** [comment]
@MusounoEnma Please familiarize with Tenstorrent bounties process.
If you want to propose a new bounty, please send an email to [bounties@tenstorrent.com](mailto:bounties@tenstorrent.com) for a review.

**Comment by MusounoEnma** [comment]
Got it, thank you for the guidance! I have sent the proposal to bounties@tenstorrent.com, Thanks again for the review and support on this PR.
